pkg/libartifact/artifact.go

```diff
@@ -11,17 +11,15 @@ import (
 )
 
 type Artifact struct {
-	Manifests []manifest.OCI1
-	Name      string
+	Manifest *manifest.OCI1
+	Name     string
 }
 
 // TotalSizeBytes returns the total bytes of the all the artifact layers
 func (a *Artifact) TotalSizeBytes() int64 {
 	var s int64
-	for _, artifact := range a.Manifests {
-		for _, layer := range artifact.Layers {
-			s += layer.Size
-		}
+	for _, layer := range a.Manifest.Layers {
```

Comment:
This is not blocking; a segfault will occur if the `Manifest` is `nil`. Should the caller or the method check this?

Reply:
Do we have to? The only way this can end up nil is if
`mani, err := manifest.OCI1FromManifest(b)`, returns nil,nil which should not be possible.

It may be good to check for nil in places where we cannot be sure but IMO if Manifest ends up being nil in any code path we a have big problem that might as well panic.

Reply:
Ok, Thanks

Reply:
Consider either making the field a non-pointer, or at least explicitly documenting that is guaranteed not to be `nil`. (And I will be griping about no nullability in the type system.)

Reply:
Or, alternatively, if this is a forward-compatibility mechanism where hypothetically future artifacts might _not_ have a single manifest, that definitely needs to be documented (something to the effect of "current implementation never has `nil`, but that may happen in the future, so _external_ consumers must be ready to handle that").

Reply:
IMO making it not a pointer doesn't help us, sure it will not panic in such a place but it doesn't change the fact that the struct would hold invalid manifest data (zero type) then. Because `schemaVersion` 0 or an empty `mediaType` are clearly worthless and would just likely cause logic bugs in consumers.